src/request/change-password/ChangePassword.js

```diff
@@ -0,0 +1,211 @@
+            4,
+            1,
+        );
+        this._progressIndicator = new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.SET_PASSWORD} .progress-indicator`),
+            4,
+            2,
+        );
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.DOWNLOAD_FILE} .progress-indicator`),
+            4,
+            4,
+        );
+
+        // Events
+
+        this._passwordGetter.on(PassphraseBox.Events.SUBMIT, this._unlock.bind(this));
+
+        this._passwordSetter.on(PassphraseSetterBox.Events.ENTERED, () => {
+            let colorClass = '';
+            if (this._key.secret instanceof Nimiq.Entropy) {
+                const color = Iqons.getBackgroundColorIndex(
+                    new Nimiq.Address(
+                        // use color of first address as loginFile color
+                        this._key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+                    ).toUserFriendlyAddress(),
+                );
+                const colorString = LoginFile.CONFIG[color].name;
+                colorClass = `nq-${colorString}-bg`;
+            }
+            this._loginFileIcon.lock(colorClass);
+            this._progressIndicator.setStep(3);
+        });
+
+        this._passwordSetter.on(PassphraseSetterBox.Events.SUBMIT, /** @param {string} password */ async password => {
+            const passwordBytes = Utf8Tools.stringToUtf8ByteArray(password);
+
+            await KeyStore.instance.put(this._key, passwordBytes);
+
+            if (this._key.secret instanceof Nimiq.PrivateKey) {
+                this._resolve({ success: true });
+                return;
+            }
+
+            // Prepare Login File for download
+
+            const firstAddress = new Nimiq.Address(
+                this._key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+            );
+
+            const encryptedEntropy = await this._key.secret.exportEncrypted(passwordBytes);
+
+            downloadLoginFile.setEncryptedEntropy(
+                /** @type {Nimiq.SerialBuffer} */ (encryptedEntropy),
+                firstAddress,
+            );
+
+            downloadLoginFile.on(DownloadLoginFile.Events.DOWNLOADED, () => {
+                this._resolve({ success: true });
+            });
+            window.location.hash = ChangePassword.Pages.DOWNLOAD_FILE;
+        });
+        this._passwordSetter.on(PassphraseSetterBox.Events.NOT_EQUAL, () => this._loginFileIcon.unlock());
+    }
+
+    async run() {
+        this._passwordGetter.reset();
+        window.location.hash = ChangePassword.Pages.ENTER_PASSWORD;
+        if (TopLevelApi.getDocumentWidth() > Constants.MIN_WIDTH_FOR_AUTOFOCUS) {
+            this._passwordGetter.focus();
+        }
+    }
+
+    /**
+     * @param {string} password
+     */
+    async _unlock(password) {
```

Comment:
The parameter should be named `oldPassword` instead, to differentiate between the passwords (for code clarity).